src/pages/WalletManagement/WalletWordsPage.tsx

```diff
@@ -34,10 +36,17 @@ import { useStepsContext } from '../../contexts/steps'
 import { useWalletContext } from '../../contexts/wallet'
 
 const WalletWordsPage = () => {
-  const { mnemonic, plainWallet } = useWalletContext()
+  const { mnemonic, plainWallet, setPlainWallet, setMnemonic } = useWalletContext()
   const { onButtonBack, onButtonNext } = useStepsContext()
   const { setSnackbarMessage } = useGlobalContext()
 
+  // Init wallet
```

Comment:
I moved the actual wallet generation AFTER the account creation step. If the user never ends up creating an account, there's no need creating a wallet and setting it to our app's state. This also helps to clean up our routes component, so that it's only responsible for the routes.